### DISCUSSION TITLE
PS/2 Keyboard w/ USB Interface

### DISCUSSION THREAD

**Discussion by VanKichline** [thehighnibble/thehighnibble.github.io#167] [S-132 add-on for IMSAI & Cromemco replicas]
I've assembled my S-132 and am very happy with it. Excellent design and implementation, makes the kit even more enjoyable.
When I ordered, I requested the USB interface, but since then I've adopted an Agon Light 2 which insists on a PS/2 keyboard. I found a really nice KeyTronics keyboard, and I have an ancient 2 station PS2/VGA KVM which works well.
Unfortunately, I haven't found a PS/2 to USB adaptor that works with the S-132.  All is well when using a USB keyboard on the S-132, but then there are two keyboards on my desk. I have tried two smart PS/2>USB adaptors and one pass-through model (none of which have a brand name or model number,) and none work at all with the S-132, by which I mean the keyboard does not seem to be attached.  The video switches as expected.
Does anyone have a PS/2 adaptor recommendation? I could also desolder the USB board and add a "Mini-DIN-6 PS/2 Keyboard Female Socket - Right Angled," but I'm concerned about finding a mountable part. Could anyone provide a physically compatible part number and supplier?
Thanks!

**Comment by VanKichline** [comment]
I retract, and apologize.  After posting, I realized I had not tested the keyboard and adaptor plugged directly into the S-132, and it turns out this works fine.  So it appears to be a problem with the KVM.  I don't understand why it passed my tests of its two channels, but those tests did not include the adaptor/converter.
I think finding a working PS/2 VGA KVM will be the difficult part. Worth another trip to the attic, at least.

**Comment by s-moores** [comment]
I also connected my S-132 to my primary KVM switch and found it would not work. I'm guessing that perhaps some code in the CH559 doesn't like something about my KVM switch which does present not only as a simple keyboard but also as a USB mouse/mass storage and printer. I've been considering trying a [HID Keyboard remapper](https://github.com/jfedor2/hid-remapper) to see if I can use that to filter out whatever comes from my KVM switch (beyond a simple keyboard) to allow it to work but haven't got there yet (there just aren't enough hours in the day it seems).

**Comment by dmcnaugh** [comment]
The USB-HID adapter for both the S-132 and the VT132 only supports a USB HID keyboard device, it wont work with a USB HUB or a composite USB device. From what you are saying the KVM is presenting as either a HUB or a composite device.
